### DISCUSSION TITLE
better visibility of rivers

### DISCUSSION THREAD

**Discussion by Go77** [XCSoar/XCSoar#1364] [General]
Currently rivers are drawn as a very narrow (1 px wide?) blue line on the map. Therefore they are hardly visible. Nevertheless, they are important navigation landmarks. Is there any way to increase the width of rivers on the map?

**Comment by rwillia** [comment]
The width is determined by your map data fille, specifically the pen width value in the topology.tpl file within the map data file. You can edit this value yourself by unzipping the XCM file, editing the appropriate line in the topology.tpl file in a text editor, and zipping the XCM contents back together. Refer to the documentation here: https://xcsoar.readthedocs.io/en/latest/mapfile.html

**Reply by Go77** [reply]
Thanks a lot, rwillia, this is useful information! I looked into the 'waterline'-files to find this option but overlooked the topology-file. Works like a charm.

Maybe I can also discover now why a number of the large Dutch polders in the IJsselmeer are drawn as water-areas instead of land... It is strange that some polders are correctly drawn as land and others as water-areas.

**Comment by lordfolken** [comment]
> Maybe I can also discover now why a number of the large Dutch polders in the IJsselmeer are drawn as water-areas instead of land... It is strange that some polders are correctly drawn as land and others as water-areas.

That area is problematic because parts of it are tagged as lake, and other parts as sea, some of it coastline.

Currently the mapgen substracs the sea parts from the terrain file. Water bodies are polygons in the topology file. 
This process uses the the waterpolygons from here: https://mapgen-data.sigkill.ch/waterpolygons

At the moment we use the [hydrolakes](https://www.hydrosheds.org/products/hydrolakes) dataset for lakes. 

The current plan is to replace with OSM data, as especially river polygons are currently missing.

I have written an extractor for most topology data, with which i hope to replace all the map data. You can use localized osm data with it if you want to not deal with hughe data. https://github.com/lordfolken/xcsoar-mapgen-topology

Some examples of that you can see here: https://forum.xcsoar.org/viewtopic.php?t=4281

**Reply by Go77** [reply]
Dear lordfolken, thank you for all the work you're putting in it. I had a look at the examples and they look great. I eagerly await the update of mapgen!

**Reply by lordfolken** [reply]
I would welcome more people experimenting and potentionally contribute their findings. 😃
